dot-studio/protobuf

```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+# shellcheck disable=SC2154,SC1091
+#
+# Copyright:: Copyright 2017 Chef Software, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+document "compile_go_protobuf" <<DOC
+  Compile the protobuf definitions from the provided script.
+
+  @(arg:1) Path to the script to execute. (default: scripts/grpc.sh)
+
+  The script should look like:
+  ------------------------------------------------------------------
+  #!/bin/bash
+  set -x
+
+  protoc -I. \\
+    -I$GOPATH/src \\
+    -I$GOPATH/src/github.com/grpc-ecosystem/grpc-gateway/third_party/googleapis \\
+    --go_out=plugins=grpc:. \\
+    inspec/*.proto
+  ------------------------------------------------------------------
+
+DOC
+function compile_go_protobuf() {
+  local proto_go_tools=(
+    github.com/grpc-ecosystem/grpc-gateway/protoc-gen-grpc-gateway
+    github.com/grpc-ecosystem/grpc-gateway/protoc-gen-swagger
+    github.com/golang/protobuf/protoc-gen-go
+  )
+  proto_script="scripts/grpc.sh"
```

Comment:
Rather than doing this, you _should_ be able to do `proto_script="${1:-scripts/grpc.sh}"`